File: backend/src/main/java/com/ballpen/mes/service/ReportService.java
package com.ballpen.mes.service;

import com.ballpen.mes.dto.*;

import java.util.List;

public interface ReportService {

    ReportSummaryDto getExecutiveSummary();

    List<ProductionOrderDto> getProductionReport();

    List<QualityCheckDto> getQualityReport();

    List<FinishedGoodsDto> getInventoryReport();

    List<DispatchDto> getDispatchReport();
}

File: frontend/src/services/reportService.js
import api from './api';

const reportService = {
  getExecutiveSummary: async () => {
    return await api.get('/reports/executive-summary');
  },

  getProductionReport: async () => {
    return await api.get('/reports/production');
  },

  getQualityReport: async () => {
    return await api.get('/reports/quality');
  },

  getInventoryReport: async () => {
    return await api.get('/reports/inventory');
  },

  getDispatchReport: async () => {
    return await api.get('/reports/dispatch');
  },
};

export default reportService;
